app/assets/javascripts/directives/require_file.js

```diff
@@ -0,0 +1,24 @@
+ManageIQ.angular.app.directive('requireFile', ['$parse', function ($parse) {
```

Comment:
Yes, it does parse the uploaded file to string. I'm not sure if I can get the file info without looking at the element though, because Angular will not bind file inputs to ng-model as far as I understand. I'm sure I could put the on-change function in the controller, but I am thinking it would still have to look at the element because it doesn't bind the model. I may be mistaken though, I'm still fairly unfamiliar with Angular.